File: leetcode/Accepted/range-sum-of-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
  private:
    int sum = 0;
  public:
    int rangeSumBST(TreeNode* root, int low, int high) {
      if(!root) return 0;
      if(root->val >= low and root->val <= high) sum += root->val;
      if(root->val > low) rangeSumBST(root->left, low, high);
      if(root->val < high) rangeSumBST(root->right, low, high);
      return sum;
    }
};

File: leetcode/Easy/range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        def dfs(node):
            if node:
                if low <= node.val <= high:
                    self.ans += node.val

                if low < node.val:
                    dfs(node.left)

                if high > node.val:
                    dfs(node.right)

        self.ans = 0
        dfs(root)
        return self.ans
